core-processor/src/processor.rs

```diff
@@ -127,25 +127,35 @@ pub fn prepare(
     }
 
     let mut gas_allowance_counter = GasAllowanceCounter::new(gas_allowance);
-    let memory_size = match executor::charge_gas_for_pages(
-        &block_config.allocations_config,
-        &mut gas_counter,
-        &mut gas_allowance_counter,
-        program.get_allocations(),
-        program.static_pages(),
-        dispatch.context().is_none() && matches!(dispatch.kind(), DispatchKind::Init),
-        subsequent_execution,
-    ) {
+    let mut f = || {
+        let memory_size = executor::charge_gas_for_pages(
+            &block_config.allocations_config,
+            &mut gas_counter,
+            &mut gas_allowance_counter,
+            program.get_allocations(),
+            program.static_pages(),
+            dispatch.context().is_none() && matches!(dispatch.kind(), DispatchKind::Init),
+            subsequent_execution,
+        )?;
+        executor::charge_gas_for_instantiation(
+            block_config.module_instantiation,
```

Comment:
how can you charge some "constant" to block value, if instantiation cost is proportional to code size?

there should be param of code len and cost per byte, than `charge_gas_for_instantiation` calculates it automatically